Content: update help channel guide

Akarys42 opened the pull request:
This PR changes the following inside the help channel guide:
* Only one help channel can be claimed at the same time
* You can use the search to find your channel
* The channel will close after 10 minutes if someone else sends a message

Akarys42 committed:
Content: update help channel claiming system

This commit changes the following inside the help channel guide:
* Only one help channel can be claimed at the same time
* You can use the search to find your channel
* The channel will close after 10 minutes if someone else sends a message

TizzySaurus commented on pydis_site/apps/content/resources/guides/pydis-guides/help-channel-guide.md:
```diff
@@ -42,14 +42,14 @@ There are always 3 available help channels waiting to be claimed in the **Python
 
 In order to claim one, simply start typing your question into one of these channels. Once your question has been posted, you have claimed this channel, and the channel will be moved down to the **Python Help: Occupied** category.
```
> ...will be moved down to the **Python Help: Occupied** category.

We've renamed the category to "Occupied Help Channels". We've also changed `Python Help: Available` on L39 etc. to `Available Help Channels` (as per the image) (`Python Help: Dormant` wasn't changed so that's still good).

Another minor thing is the fact we've now added emojis to the topics, so we may want to update the images?

sue-m-oh replied:
I've updated category names where applicable and also have updated screenshots.

TizzySaurus commented on pydis_site/apps/content/resources/guides/pydis-guides/help-channel-guide.md:
```diff
@@ -42,14 +42,14 @@ There are always 3 available help channels waiting to be claimed in the **Python
 In order to claim one, simply start typing your question into one of these channels. Once your question has been posted, you have claimed this channel, and the channel will be moved down to the **Python Help: Occupied** category.
 
-If you're unable to type into these channels, this means you're currently **on cooldown**. In order to prevent someone from claiming all the channels for themselves, **we only allow someone to claim a new help channel every 15 minutes**. However, if you close your help channel using the `!dormant` command, this cooldown is reset early.
+If you're unable to type into these channels, this means you're currently **on cooldown**. In order to prevent someone from claiming all the channels for themselves, **we only allow someone to claim a single help channel at the same time**. You can search in the top right corner of your Discord client `from:yourusername#xxxx` to find back your channel.
```
>> **we only allow someone to claim a single help channel at the same time**

Perhaps reword this bold part to "we only allow a user to have a single help channel at any time."

>> You can search in the top right corner of your Discord client `from:yourusername#xxxx` to find back your channel.

Perhaps also mention the "new" `!helpdm` feature, which serves a similar purpose.

sue-m-oh replied:
I believe these are addressed in the re-write. Please take a look!

`!helpdm` mentioned in https://github.com/python-discord/site/pull/624/commits/32db9c54f46edbca8036f6cf2e7dc94ac78ad056

TizzySaurus commented on pydis_site/apps/content/resources/guides/pydis-guides/help-channel-guide.md:
```diff
@@ -42,14 +42,14 @@ There are always 3 available help channels waiting to be claimed in the **Python
 ## Q: For how long is the channel mine?
 
-The channel is yours until it has been inactive for **30 minutes**. When this happens, we move the channel down to the **Python Help: Dormant** category, and make the channel read-only. After a while, the channel will be rotated back into **Python Help: Available** for the next question. Please try to resist the urge to continue bumping the channel so that it never gets marked as inactive. If nob ...
+The channel is yours until it has been inactive for **10 minutes**, or 30 minutes until someone participate in the channel. When this happens, we move the channel down to the **Python Help: Dormant** category, and make the channel read-only. After a while, the channel will be rotated back into **Python Help: Available** for the next question. Please try to resist the urge to continue bumping the  ...
```
> The channel is yours until it has been inactive for **10 minutes**, or 30 minutes until someone participate in the channel.

This doesn't read very well to me. Perhaps "The channel is yours until either 30 minutes of no activity from your opening message or 10 minutes of no activity from the last message."

It *may* be "last comment *from the claimant*, not sure.

sue-m-oh replied:
This wording was fixed in https://github.com/python-discord/site/pull/624/commits/32db9c54f46edbca8036f6cf2e7dc94ac78ad056

TizzySaurus commented on pydis_site/apps/content/resources/guides/pydis-guides/help-channel-guide.md:
```diff
@@ -42,14 +42,14 @@ There are always 3 available help channels waiting to be claimed in the **Python
-The channel is yours until it has been inactive for **30 minutes**. When this happens, we move the channel down to the **Python Help: Dormant** category, and make the channel read-only. After a while, the channel will be rotated back into **Python Help: Available** for the next question. Please try to resist the urge to continue bumping the channel so that it never gets marked as inactive. If nob ...
+The channel is yours until it has been inactive for **10 minutes**, or 30 minutes until someone participate in the channel. When this happens, we move the channel down to the **Python Help: Dormant** category, and make the channel read-only. After a while, the channel will be rotated back into **Python Help: Available** for the next question. Please try to resist the urge to continue bumping the  ...
 
 ![Channel dormant message](/static/images/content/help_channels/dormant_message.png)
```
This image isn't correct as it shows the category as `Help: Dormant` instead of `Python Help: Dormant`. To be consistent with the `Channel available message` image it should probably include the emoji too (just need to re-take the screenshot).

Separate point but GitHub won't let me make it on the correct line:
Line 59 has a splice comma (the "You will only be able to run the command if you claimed the channel from the available category, you cannot close channels belonging to others". This `,` should be a `;`.

sue-m-oh replied:
Said screenshot was updated in https://github.com/python-discord/site/pull/624/commits/32db9c54f46edbca8036f6cf2e7dc94ac78ad056

The area around previous line 59 has been re-written in that same above commit on line 48.

TizzySaurus commented on pydis_site/apps/content/resources/guides/pydis-guides/help-channel-guide.md:
```diff
@@ -42,14 +42,14 @@ There are always 3 available help channels waiting to be claimed in the **Python
-If you're unable to type into these channels, this means you're currently **on cooldown**. In order to prevent someone from claiming all the channels for themselves, **we only allow someone to claim a new help channel every 15 minutes**. However, if you close your help channel using the `!dormant` command, this cooldown is reset early.
+If you're unable to type into these channels, this means you're currently **on cooldown**. In order to prevent someone from claiming all the channels for themselves, **we only allow someone to claim a single help channel at the same time**. You can search in the top right corner of your Discord client `from:yourusername#xxxx` to find back your channel.
 
 ![Channel available message](/static/images/content/help_channels/available_message.png)
```
The message has since been updated, so this image should be updated to reflect the new message.

sue-m-oh replied:
Addressed in https://github.com/python-discord/site/pull/624/commits/32db9c54f46edbca8036f6cf2e7dc94ac78ad056

TizzySaurus reviewed (dismissed):
There's a few other areas that also need updating, and the odd grammatical point.

minalike committed:
Fix markdown error linking to recommended editors

minalike committed:
Update wording in topical and general help channel sections

minalike committed:
Add Frequently Asked Questions header

minalike committed:
Update channel dormancy timings

minalike committed:
Add "no one answered me" question and answer

minalike committed:
Edit "how do I close my channel" question and answer

minalike committed:
Edit "can only Helpers help?" answer and question

Remove screenshot of occupied help channels because it doesn't serve much purpose, and help can also be provided in topical channels.

MarkKoz commented on pydis_site/apps/content/resources/guides/pydis-guides/help-channel-guide.md:
```diff
@@ -22,49 +22,53 @@ If your question fits into the domain of one of our topical help channels, and i
+## Q: How long does my help channel stay active?
 
+The channel remains open for **30 minutes** after your last message, but if another user posts in your channel, the closing will be delayed by 10 minutes.
+
```
This isn't accurate. If the claimant sends a message, and then another user immediately sends a message, the channel will still close in 30 minutes. It does not get delayed by 10 minutes in such case. An accurate way to describe it is "it closes 30 minutes after your last message or 10 minutes after the last message sent by another user, whichever time comes later."

sue-m-oh replied:
I went with your wording.

MarkKoz commented on pydis_site/apps/content/resources/guides/pydis-guides/help-channel-guide.md:
```diff
@@ -22,49 +22,53 @@ If your question fits into the domain of one of our topical help channels, and i
+
+If you still feel like your question is being overlooked, read our guide on [asking good questions](../asking-good-questions) to incease your chances of getting a response.
+
+## Q: I don't need my help channel anymore, my question was answered. What do I do?
```
This is a comma splice 
```suggestion
## Q: I don't need my help channel anymore; my question was answered. What do I do?
```

sue-m-oh replied:
How about just "My question was answered. What do I do?"

MarkKoz replied:
Sure.

MarkKoz commented on pydis_site/apps/content/resources/guides/pydis-guides/help-channel-guide.md:
```diff
@@ -22,49 +22,53 @@ If your question fits into the domain of one of our topical help channels, and i
+## Q: I don't need my help channel anymore, my question was answered. What do I do?
 
-## Q: Are only Helpers supposed to answer questions?
+Go ahead and move your channel to the dormant category by running the `!closed` command. You will only be able to run this command in your own help channel, and likewise no one will be able to prematurely close your claimed help channel (with the exception of staff).
```
You omitted the explanation of each category when discussing the channel cycle in "For how long is the channel mine?" Thus, I think mentioning the dormant category here may be confusing. Do you think it's worth elaborating a bit on what being moved to that category means?

sue-m-oh replied:
I added an additional separate question that explains the 3 categories, including dormant ones.

minalike committed:
Update content

Update some screenshot images to reflect emoji used in category names
Break up some of the FAQ questions into smaller questions
Mention the !helpdm on command added in 2021
Mention the channel claimed embed added in 2022
Culling of content to focus on information pertaining to getting help

MarkKoz commented on pydis_site/apps/content/resources/guides/pydis-guides/help-channel-guide.md:
```diff
@@ -10,66 +10,84 @@ relevant_links:
+*This message indicates that a channel is available.*
 
-If you're unable to type into these channels, this means you're currently **on cooldown**. In order to prevent someone from claiming all the channels for themselves, **we only allow someone to claim a new help channel every 15 minutes**. However, if you close your help channel using the `!dormant` command, this cooldown is reset early.
+In order to claim one, simply ask your question into one of the available channels. Be sure to [ask good questions](../asking-good-questions) in order to give yourself the best chances of getting help!
```
"ask into" is awkward. It should be "ask in"

sue-m-oh replied:
Yup, will fix.

sue-m-oh replied:
[4a0a79b](https://github.com/python-discord/site/pull/624/commits/4a0a79b184b292fc2f03f0d505f854cba6b2aa0d)

MarkKoz reviewed (changes requested):
Looks good.

mbaruh commented on pydis_site/apps/content/resources/guides/pydis-guides/help-channel-guide.md:
```diff
@@ -10,66 +10,84 @@ relevant_links:
-Where you should go depends on what you need help with.
-These channels also attract different helpers, and move at different speeds, which affects the kind of help you're likely to receive, and how fast you get that help.
+We have two different kinds of help channels in our community: **topical help channels** and **general help channels**.
+The topical channels can be great for longer and broader discussions, whereas a general help channel might be more appropriate for specific troubleshooting. Individual topical channels may have a narrower audience than the general help channels, but that may also be to your advantage if the audience is very familiar with the area of Python you are struggling with.
```
I'm not sure I like this description, it's a bit confusing. If I were to describe it, I'd just say that the help system provides channels for focused help to you individually, and can have any python related question, while the topical channels have specific topics, aren't reserved for one person at a time, but might have a better chance to been seen by someone with domain knowledge.

sue-m-oh replied:
I agree, reading it now it is confusing hmmmmmmmmmm.

sue-m-oh replied:
How's this @mbaruh ?

In topical channels, many users can discuss and ask for help around specific domains or areas of Python; anyone can participate in giving and asking help. General help channels can be used for all Python-related help, and are temporarily claimed and reserved by individual users so that they can get focused attention to their problem.

sue-m-oh replied:
Removing this introduction section and adding the information to their respective sections as needed. Adding a table of contents instead.

mbaruh commented on pydis_site/apps/content/resources/guides/pydis-guides/help-channel-guide.md:
```diff
@@ -10,66 +10,84 @@ relevant_links:
+*This message indicates that a channel is available.*
 
-If you're unable to type into these channels, this means you're currently **on cooldown**. In order to prevent someone from claiming all the channels for themselves, **we only allow someone to claim a new help channel every 15 minutes**. However, if you close your help channel using the `!dormant` command, this cooldown is reset early.
+In order to claim one, simply ask your question in one of the available channels. Be sure to [ask good questions](../asking-good-questions) in order to give yourself the best chances of getting help!
```
maybe we could call it "asking questions well" or something? someone might think that their problem might not be good enough to be asked about or something.

sue-m-oh replied:
Rephrasing to "Be sure to ask questions with enough information..."

mbaruh commented on pydis_site/apps/content/resources/guides/pydis-guides/help-channel-guide.md:
```diff
@@ -10,66 +10,84 @@ relevant_links:
+*This messages indicates that you've claimed the channel.*
 
-## Q: For how long is the channel mine?
+At this point you will have the **Help Cooldown** role which will remain on your profile until your help channel is closed (with the `!close` command) or goes dormant due to inactivity. This ensures that users can claim only one help channel at any given time, giving everyone a chance to have their question seen.
```
Could say "until you close your help channel" to highlight the ability of closing the channel, even if it's not as accurate.

sue-m-oh replied:
Yes

mbaruh commented on pydis_site/apps/content/resources/guides/pydis-guides/help-channel-guide.md:
```diff
@@ -10,66 +10,84 @@ relevant_links:
+*You'll see this message in your channel once it goes dormant.*
+### Q: No one answered my question. How come?
+
+The server has users active all over the world and all hours of the day, but some time periods are less active than others. It's also possible that the users that read your question didn't have the knowledge required to help you. If no one responded, feel free to claim another help channel a little later.
```
This can also suggest asking in a topical channel if appropriate.

sue-m-oh replied:
Yes

mbaruh reviewed (commented):
Looks good, a few comments

minalike committed:
Add table of content and final edits

minalike committed:
Merge branch 'main' into content/update-help-channel-timing

sue-m-oh commented on pydis_site/apps/content/resources/guides/pydis-guides/help-channel-guide.md:
```diff
@@ -6,70 +6,86 @@ relevant_links:
-Our general help channels move at a fast pace, and attract a far more diverse spectrum of helpers.
-This is a great choice for a generic Python question, and a good choice if you need an answer as soon as possible.
-It's particularly important to [ask good questions](../asking-good-questions) when asking in these channels, or you risk not getting an answer and having your help channel be claimed by someone else.
+General help channels can be used for all Python-related help, and have the advantage of attracting a more diverse spectrum of helpers, and getting individual focus and attention on your question. These channels are a great choice for generic Python help, but can be used for domain-specific Python help as well.
```
double "and" - will reword

sue-m-oh replied:
https://github.com/python-discord/site/commit/16a9727d414ed931cc74fa49f86a52f288399fa1 "General help channels can be used for all Python-related help, and have the advantage of attracting a more diverse spectrum of helpers. There is also the added benefit of receiving individual focus and attention on your question."

jchristgit commented on pydis_site/apps/content/resources/guides/pydis-guides/help-channel-guide.md:
```diff
@@ -6,70 +6,86 @@ relevant_links:
+General help channels can be used for all Python-related help, and have the advantage of attracting a more diverse spectrum of helpers, and getting individual focus and attention on your question. These channels are a great choice for generic Python help, but can be used for domain-specific Python help as well.
 
-## How To Claim a Channel
+## How to Claim a Channel
```
should we also lowercase the rest then or was this intentional?

mbaruh replied:
Prepositions are just not capitalized in titles.

sue-m-oh replied:
Yup, intentionally changed to lowercase

jchristgit commented on pydis_site/apps/content/resources/guides/pydis-guides/help-channel-guide.md:
```diff
@@ -6,70 +6,86 @@ relevant_links:
+
+The server has users active all over the world and all hours of the day, but some time periods are less active than others. It's also possible that the users that read your question didn't have the knowledge required to help you. If no one responded, feel free to claim another help channel a little later, or try an appropriate topical channel.
+
+If you feel like your question is continuously being overlooked, read our guide on [asking good questions](../asking-good-questions) to incease your chances of getting a response.
```
[rrrrrrrrrrrrrrrrrrrrrrrr](https://www.youtube.com/watch?v=W3q8Od5qJio)

```suggestion
If you feel like your question is continuously being overlooked, read our guide on [asking good questions](../asking-good-questions) to increase your chances of getting a response.
```

sue-m-oh replied:
Yes sir
Fixed in https://github.com/python-discord/site/pull/624/commits/16a9727d414ed931cc74fa49f86a52f288399fa1

jchristgit reviewed (approved):
Absolute win. Two small comments, apart from that good to go

mbaruh reviewed (approved):
I'm not needed anymore but I still approve of this after your address Volcyy's comments. Thanks for the effort :)

minalike committed:
Fixup: final wording and typo

mbaruh committed:
Merge branch 'main' into content/update-help-channel-timing
